trunner/config.py

```diff
@@ -30,73 +37,134 @@ def __init__(self, parser, message="Parser error"):
+        self.parse_keywords(test)
+        self.parse_name(test)
+        self.parse_target(test)
         # If type is not known assume that it's a unit test
-        config.setdefault('type', 'unit')
-        config.setdefault('ignore', False)
-        config.setdefault('timeout', PYEXPECT_TIMEOUT)
+        test.setdefault('type', 'unit')
+        test.setdefault('timeout', PYEXPECT_TIMEOUT)
+
+        test.setdefault('ignore', False)
+        if not isinstance(test['ignore'], bool):
+            test['ignore'] = False
+
+        if test.get('harness'):
+            self.parse_harness(test)
+
+        # Copy test per target
+        test_case = []
+        for target in test['targets']['value']:
+            test_by_target = test.copy()
+            test_by_target['target'] = target
+            test_case.append(test_by_target)
+            del test_by_target['targets']
 
-        if not isinstance(config['ignore'], bool):
-            config['ignore'] = False
+        return test_case
 
-        if config.get('harness'):
-            config['type'] = 'harness'
-            config['harness'] = self.path / config['harness']
+    def inherit_array_value(self, test, keyword):
+        global_array = self.config.get(keyword)
+        array = test.get(keyword)
 
-            if not config['harness'].exists():
-                raise YAMLParserError(self, f"harness {config['harness']} file not found")
+        if global_array and array:
+            array.setdefault('value', global_array['value'])
 
-            if not str(config['harness']).endswith('.py'):
-                raise YAMLParserError(self, "harness file must has .py extension")
+    def inherit_global_keywords(self, test):
+        self.inherit_array_value(test, 'targets')
 
-        return config
+        for key in self.config:
+            if key not in test:
+                test[key] = self.config[key]
 
     def parse_test_config(self):
-        config = self.config
-        if 'test' not in config:
+        if 'test' not in self.config:
             raise YAMLParserError(self, 'Keyword "test" not found in test config')
 
-        config = config['test']
-        if 'tests' not in config:
+        self.config = self.config['test']
+        if 'tests' not in self.config:
             raise YAMLParserError(self, 'Keyword "tests" not found in test config')
 
-        for test_config in config['tests']:
-            # Combine 'local' and 'global' config
-            for key in config:
-                if key not in test_config and key != 'tests':
-                    test_config[key] = config[key]
+        if 'targets' in self.config:
+            self.parse_target(self.config)
 
-            test = self.parse_test_case(test_config)
-            self.parsed_tests.append(test)
+        tests_configs = self.config.pop('tests')
+        for test_config in tests_configs:
+            # Combine 'local' test config and 'global' config
+            self.inherit_global_keywords(test_config)
```

Comment:
small note - the naming `inherit_global_keywords` might be misleading, because it's not `self` that is inheriting some properties from the `test_config` - it's the other way around. So maybe it should be `propagate` or something like that.

Might be changed in the future, let's leave it as-is for now.